### ISSUE TITLE
Clarification on structuring inputs for partially labeled datasets and semi-supervised learning tasks

### ISSUE THREAD

**Issue by faris-k** [lightly-ai/lightly#877]
As per the tutorial on [structuring your inputs](https://docs.lightly.ai/tutorials/structure_your_input.html#directory-with-subdirectories-containing-images), subdirectories in your input directory get assigned a "weak label." If we have a dataset with both labeled samples and unlabeled samples, is this the correct way to go about structuring the data if we're approaching a semi-supervised task? 

- data/
    - dogs/
        - lab.png
        - poodle.png
    - cats/
        - siamese.png
        - tabby.png
    - unlabeled/
        - unknown-1.png
        - unknown-2.png
        - ...
        - unknown-99.png

I'm a bit confused what "weak" labels mean here. Let's say that the `dogs` and `cats` directories correspond to "strong" labels, i.e. images of cats and dogs that have been hand-labeled. If the `unlabeled` directory consists of images of dogs and cats that I haven't labeled, does my input directory scheme imply that these images belong to a third, separate class? Should the unlabeled samples instead go in the main directory instead of subdirectories, as below?

- data/
    - dogs/
        - lab.png
        - poodle.png
    - cats/
        - siamese.png
        - tabby.png
    - unknown-1.png
    - unknown-2.png
    - ...
    - unknown-99.png

**Comment by guarin** [comment]
Hi @faris-k!

The name of the "weak labels" might be a bit confusing. The "weak" just comes from the fact that classes are assigned based on the subdirectory (or video for video datasets). We don't make a difference between "strong" and "weak" labels.

For a semi-supervised task I would suggest to put all unlabelled images in a common subdirectory as in your first example. During training you can then simply ignore the images with the "unlabelled" class for the classification task. If you do pretraining on all images and finetuning only on the labeled images you could structure your data as follows:

* data/
  *  labelled/
     * dogs/
     * cats/
     * ...
  * unlabelled/
     * unknown-1.png
     * unknown-2.png
     * ...

And load it with this code:
```
pretrain_dataset = LightlyDataset("data")
finetune_dataset = LightlyDataset("data/labelled")
```

Hope this helps and sorry for the late reply!
